lsp-protocol.el

```diff
@@ -111,12 +111,20 @@ Example usage with `dash`.
                                                ',required))))
                     `(cl-defun ,(intern (format "lsp-make-%s" (s-dashed-words (symbol-name interface))))
                          (&rest plist &key ,@(-map (-lambda ((key))
-                                                     (intern (substring (symbol-name key) 1))) params)
+                                                     (let ((key-sym (intern (substring (symbol-name key) 1))))
+                                                       (if (special-variable-p key-sym)
+                                                           `((,key ,(intern (format "%s_" (symbol-name key-sym)))))
+                                                         key-sym)))
+                                                     params)
                                 &allow-other-keys)
-                       (ignore ,@(-map (-lambda ((key))
-                                         (intern (substring (symbol-name key) 1))) params))
```

Comment:
FYI, I have reverted L116 and L117 in 73ad089abf55ac946a6776ffc431eac31627748d to prevent more compile warnings.

Reply:
This change so that we have the correct doc string for those function. Reverting them will make the docstring turn out to be wrong :(

Can we disable the docstring length warning instead? That would be hard to keep, especially with auto generated function from lsp-interface macro?

Reply:
Ah, I got it. I think there is a variable to configure to turn off the 80-character limit. 🤔

Let me see what I can do about this! 👌

Reply:
Probably not the best solution, but I've raised the 80-character limit to 1000. 🤔